File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    // 박스 포장 문제와 같음
    // 100 - progresses[i] = 7, 70, 45
    // (7, 70, 45) / speeds[i] = 7, Math.ceil(2) = 3, 9
    // newArr.push([7,3,9])
    
    // 첫번 째 수보다 크면 그게 max,
    // result.push(newArr.splice(0, max)) => [2,1]
    
    let rest = progresses.map(i => 100 - i)
    let newArr = rest.map((i,idx) => Math.ceil(i / speeds[idx]))
    console.log(newArr)
    let max = 0;
    let result = []
    while(newArr.length !== 0){
        // 첫번째 요소보다 큰 수면 만족하는 첫 번째 요소 인덱스 반환, 아니면 -1
        let n = newArr.findIndex(e => newArr[0] < e)
        console.log(n)
        if(n === -1){
            if(newArr.length > max){
                max = newArr.length                
            }
            result.push(newArr.length)
            newArr = []
            
        } else {
            if(n > max){
                max = n
            }
            result.push(newArr.splice(0, n).length)
        }
    }
    return result
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    # 100 - progresses / speeds
    restProg = list(map(lambda x: math.ceil((100 - progresses[x]) / speeds[x]), range(len(progresses))))
    
    count = 1
    newArr = []
    
    for i in range(len(restProg)):
        try:
            # 이전 인덱스가 작으면 count = 1
            if restProg[i] < restProg[i+1]:
                newArr.append(count)
                count = 1
            # 아니라면 count += 1
            else:
                restProg[i+1] = restProg[i]
                count += 1            
        except IndexError:
            newArr.append(count)
    return newArr